app/models/ext_management_system.rb

```diff
@@ -150,6 +150,10 @@ def self.with_hostname(hostname)
     joins(:endpoints).where(:endpoints => {:hostname => hostname})
   end
 
+  def self.with_role(role)
+    joins(:endpoints).where(:endpoints => {:role => role})
+  end
+
```

Comment:
:cry: Should be a scope...but I see you're just following what's already here :+1: